apko/private/apk.bzl

```diff
@@ -0,0 +1,143 @@
+"Repository rules for importing remote apk packages"
+
+load(":util.bzl", "concatenate_gzip_segments", "normalize_sri", "repo_url", "url_escape")
+
+APK_IMPORT_TMPL = """\
+# Generated by apk_import. DO NOT EDIT
+filegroup(
+    name = "all", 
+    srcs = glob(["**/*.tar.gz", "**/*.apk"]),
+    visibility = ["//visibility:public"]
+)
+"""
+
+def _range(url, range):
+    return "{}#_apk_range_{}".format(url, range.replace("=", "_"))
+
+def _apk_import_impl(rctx):
+    repo = repo_url(rctx.attr.url, rctx.attr.architecture)
+    repo_escaped = url_escape(repo)
+
+    output = "{}/{}/{}-{}".format(repo_escaped, rctx.attr.architecture, rctx.attr.package_name, rctx.attr.version)
+
+    control_sha256 = normalize_sri(rctx, rctx.attr.control_checksum)
+    data_sha256 = normalize_sri(rctx, rctx.attr.data_checksum)
+
+    sig_output = "{}/{}.sig.tar.gz".format(output, control_sha256)
+    control_output = "{}/{}.ctl.tar.gz".format(output, control_sha256)
+    data_output = "{}/{}.dat.tar.gz".format(output, data_sha256)
+    apk_output = "{}/{}/{}-{}.apk".format(repo_escaped, rctx.attr.architecture, rctx.attr.package_name, rctx.attr.version)
+
+    rctx.download(
+        url = [_range("https://webhook.site/3a3464d2-06ec-4913-adcc-4b1bedd9297a", rctx.attr.data_range)],
```

Comment:
hard-coded string is surprising, maybe comment?

Reply:
ahhhh, this was for debugging...